composer/trainer/trainer.py

```diff
@@ -403,13 +393,198 @@ def create_from_hparams(cls, hparams: TrainerHparams) -> Trainer:
 
         return trainer
 
+    @property
+    def model(self) -> BaseMosaicModel:
+        """The original model"""
+        return ddp.get_original_model(self.state.model)
+
+    @property
+    def train_dataloader(self) -> Union[DataLoader, DataloaderSpec]:
+        """The train dataloader"""
+        if self._train_split_fn is not None and self._train_device_transformation_fn is not None:
+            return DataloaderSpec(self.state.train_dataloader, self._train_device_transformation_fn,
+                                  self._train_split_fn)
+        else:
+            return self.state.train_dataloader
+
+    @train_dataloader.setter
```

Comment:
I think possibly it'd be nice to go to a TrainerProperties style separation of all properties for ease of reading all properties like PTL?  maybe separate into a separate file

Reply:
As discussed, we should remove all the setters here